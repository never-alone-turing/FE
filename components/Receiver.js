import React, { Component } from "react";
import { View, Text, StyleSheet, SafeAreaView } from "react-native";
import { Header } from 'react-native-elements';
import { ScrollView } from "react-native-gesture-handler";
import Checkin from './Checkin';
import normalize from 'react-native-normalize';
import {fetcher} from '../API/API.js';


class Receiver extends Component {
  constructor() {
    super();
    this.state = {
      timers: []
    }
  }

  componentDidMount = async() => {
    await this.updateTimes()
  }

  updateTimes = async() => {
    try {
      const timers = await fetcher.allTimers()
      this.setState({ timers: timers.['allCheckins'] })
    } catch(error) {
      console.log("error", error)
    }
  }

  checkIn = async(id) => {
    await fetcher.checkIn(id)
    this.updateTimes()
  }

  render () {
    return (
      <SafeAreaView>
      <View style={styles.careReceiver}>
          <Header
            containerStyle={{ backgroundColor: '#483D8B', justifyContent: 'space-around', paddingBottom: normalize(10) }}
            centerComponent={{ text: 'Never Alone', style: { color: '#fff', fontSize: normalize(30), fontWeight: "bold" } }}
            leftComponent={{ icon: 'home', color: '#fff', size: normalize(40), onPress:() => this.props.navigation.navigate('Home')} }
          />
        <ScrollView snapToAlignment="center" decelerationRate="fast" horizontal >
          {this.state.timers.map((task) => {
            return <View style={styles.task} key={task.id}>
              <Checkin checkIn={this.checkIn} task={task}/>
              <View>
                <Text style={styles.taskCategory}>{task.name}</Text>
                <Text style={styles.taskTime}>Time: {new Date(task.time).getHours().toString()}:{new Date(task.time).getMinutes().toString()}</Text>
              </View>
            </View>
          })}
        </ScrollView>
      </View>
      </SafeAreaView>
    )
  }
}

const styles = StyleSheet.create({
  careReceiver: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    textAlign: "center",
  },
  listItem: {
    padding: 15,
    backgroundColor: '#f8f8f8',
    borderBottomWidth: 1,
    borderColor: "#eee"
  },
  listItemView: {
    flexDirection: 'row'
  },
  listItemText: {
    fontSize: 18
  },
  task: {
    margin: normalize(30),
    padding: normalize(30),
    borderWidth: normalize(3),
    borderColor: "black",
    borderRadius: normalize(10),
    backgroundColor: '#E0E0E0',
  },
  taskCategory: {
    fontSize: normalize(40),
    alignSelf: 'center'
  },
  taskTime: {
    fontSize: normalize(30),
    alignSelf: 'center',
  }
})

export default Receiver;
