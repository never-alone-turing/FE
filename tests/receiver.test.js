import React from 'react'
import {create} from 'react-test-renderer'
import Receiver from '../components/Receiver'
import renderer from 'react-test-renderer'


  const mockState = {
      timers: [
        {id: 1, category: "Wake-up", time: '7:00', window: '0:15', completed: "false"},
        {id: 2, category: "Take meds", time: '12:00', window: '1:00'},
        {id: 3, category: "go to sleep", time: '22:00', window: '1:00'}
      ]
    }

  describe("<Receiver />", () => {

    it('renders correctly', () => {
      const tree = renderer.create(<Receiver
        state ={mockState}
        />).toJSON();
    expect(tree).toMatchSnapshot()
    })

  })
