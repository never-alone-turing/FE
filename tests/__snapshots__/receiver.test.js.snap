// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Receiver /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderBottomColor": "#f2f2f2",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "paddingBottom": 10,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
      testID="headerContainer"
    >
      <RNCSafeAreaView
        edges={
          Array [
            "left",
            "top",
            "right",
          ]
        }
        style={
          Object {
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 3,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#fff",
                "fontSize": 30,
                "fontWeight": "bold",
              }
            }
            text="Never Alone"
          >
            Never Alone
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
            }
          }
        />
      </RNCSafeAreaView>
    </View>
    <RCTScrollView
      collapsable={false}
      decelerationRate="fast"
      horizontal={true}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      snapToAlignment="center"
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#E0E0E0",
              "borderColor": "black",
              "borderRadius": 10,
              "borderWidth": 3,
              "margin": 30,
              "padding": 30,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#69c986",
                "borderColor": "#69c986",
                "borderRadius": 150,
                "borderWidth": 10,
                "display": "flex",
                "height": 300,
                "justifyContent": "center",
                "marginBottom": 40,
                "opacity": 1,
                "width": 300,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 35,
                    "textAlign": "center",
                  }
                }
              >
                Hold to complete task.
              </Text>
            </View>
          </View>
          <View>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 40,
                }
              }
            >
              Wake-up
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                }
              }
            >
              7:00
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#E0E0E0",
              "borderColor": "black",
              "borderRadius": 10,
              "borderWidth": 3,
              "margin": 30,
              "padding": 30,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#69c986",
                "borderColor": "#69c986",
                "borderRadius": 150,
                "borderWidth": 10,
                "display": "flex",
                "height": 300,
                "justifyContent": "center",
                "marginBottom": 40,
                "opacity": 1,
                "width": 300,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 35,
                    "textAlign": "center",
                  }
                }
              >
                Hold to complete task.
              </Text>
            </View>
          </View>
          <View>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 40,
                }
              }
            >
              Take meds
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                }
              }
            >
              12:00
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#E0E0E0",
              "borderColor": "black",
              "borderRadius": 10,
              "borderWidth": 3,
              "margin": 30,
              "padding": 30,
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#69c986",
                "borderColor": "#69c986",
                "borderRadius": 150,
                "borderWidth": 10,
                "display": "flex",
                "height": 300,
                "justifyContent": "center",
                "marginBottom": 40,
                "opacity": 1,
                "width": 300,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 35,
                    "textAlign": "center",
                  }
                }
              >
                Hold to complete task.
              </Text>
            </View>
          </View>
          <View>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 40,
                }
              }
            >
              go to sleep
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                }
              }
            >
              22:00
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
